fmp_data/lc/__init__.py

```
@@ -82,80 +72,80 @@
-                endpoints_dict[name] = (
-                    endpoint,
-                    group_config["semantics_map"][semantic_name],
-                )
+    registry = EndpointRegistry(client)
 
-        if endpoints_dict:
-            registry.register_batch(endpoints_dict)
+    # Populate registry with endpoints
+    for group_name, group_config in ENDPOINT_GROUPS.items():
+        for endpoint_name, endpoint in group_config["endpoint_map"].items():
+            registry.register_endpoint(
```

Pylint (3.3.7)

[error] Instance of 'EndpointRegistry' has no 'register_endpoint' member

(E1101)